marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.mjs

```diff
@@ -408,12 +409,34 @@ function validateAndTestMapping(mapping, uri) {
   //modify the document instance with the properly updated one if a pre-step interceptor was used
   let newDocumentIsLoad = false;
   if (mapping.name) {
+    let docActual = cts.doc(uri);
     const updatedDocument = invokeGetDocument(mapping.name, uri);
+    sourceDocument = cts.doc(uri).toObject();
```

Comment:
this can be docActual.toObject(); We don't want to call cts.doc more than necessary.